### PR TITLE
Type `PyInstaller.building.api` and related modules

### PR DESCRIPTION
**Pull request by Avasam**
During #9721 I noticed that `PyInstaller.building.api` is probably part of the public API.
And indeed this was missed in my initial `PyInstaller` PR.

Typed `PyInstaller.building.api` and related modules found in https://pyinstaller.org/en/stable/spec-files.html
Added comments to link to PyInstaller's documentation
Relaxed some `stubtest_allowlist.txt` entries so it could test a bit more of what **is** typed and/or public.

### CODE REVIEW

File: stubs/pyinstaller/PyInstaller/building/api.pyi [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+
+from PyInstaller.building import _PyiBlockCipher
+from PyInstaller.building.build_main import Analysis
+from PyInstaller.building.datastruct import TOC, Target, _TOCTuple
+from PyInstaller.building.splash import Splash
+from PyInstaller.utils.win32.versioninfo import VSVersionInfo
+from PyInstaller.utils.win32.winmanifest import Manifest
+
+if sys.platform == "linux":
+    _TargetArch: TypeAlias = Literal["x86_64", "arm64", "universal2"]
```
Comment (JelleZijlstra) [inline stubs/pyinstaller/PyInstaller/building/api.pyi:19]: Isn't universal2 a Mac thing?

File: stubs/pyinstaller/PyInstaller/building/api.pyi [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+    pkgname: str
+    toc: TOC
+    pkg: PKG
+    dependencies: TOC
+    exefiles: TOC
+    def __init__(
+        self,
+        *args: Iterable[_TOCTuple] | PYZ | Splash,
+        exclude_binaries: bool = False,
+        bootloader_ignore_signalss: bool = False,
```
Comment (JelleZijlstra) [inline stubs/pyinstaller/PyInstaller/building/api.pyi:121]:
```suggestion
        bootloader_ignore_signals: bool = False,
```

File: stubs/pyinstaller/PyInstaller/building/api.pyi (hunk 1 above)
Comment (JelleZijlstra) [reply stubs/pyinstaller/PyInstaller/building/api.pyi:19]: Indeed the EXE constructor allows non-None target arch on Mac only, not Linux.

File: stubs/pyinstaller/PyInstaller/building/api.pyi [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+        resources: Sequence[str] = ...,
+        strip: bool = False,
+        upx_exclude: Iterable[str] = ...,
+        runtime_tmpdir: str | None = None,
+        append_pkg: bool = True,
+        uac_admin: bool = False,
+        uac_uiaccess: bool = False,
+        argv_emulation: bool = False,
+        target_arch: _SuportedTargetArchParam = None,
+        codesign_identity: _CodesignIdentiyParam,
```
Comment (JelleZijlstra) [inline stubs/pyinstaller/PyInstaller/building/api.pyi:139]:
```suggestion
        codesign_identity: _CodesignIdentiyParam = None,
```
Also the type has a typo "Identiy"

File: stubs/pyinstaller/PyInstaller/utils/win32/versioninfo.pyi [hunk 4]
```diff
@@ -0,0 +1,51 @@
+from _typeshed import SliceableBuffer
+from collections.abc import Sequence
+from typing import Protocol
+from typing_extensions import Literal, TypeAlias
+
+_FourIntSequence: TypeAlias = Sequence[int] | tuple[int, int, int, int]
```
Comment (JelleZijlstra) [inline stubs/pyinstaller/PyInstaller/utils/win32/versioninfo.pyi:6]: Not sure how useful this is as a `tuple[int, int, int, int]` is a subtype of `Sequence[int]`. What type safety does this give us?

**Review by AlexWaygood — COMMENTED** [review]
Looks like I started a review of this ages ago but never finished it — here's the partial review, with a few additional points to what Jelle said :)

**Comment by Avasam** [comment]
(I've been quite occupied last week, and I got sick this week, will go through reviews soon once I feel a bit better)

**Review by JelleZijlstra — APPROVED** [review]

**Pull request merged**